db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb

```diff
@@ -4,46 +4,50 @@ class MigrateInteractionSettingsToPolicy < ActiveRecord::Migration[7.1]
   disable_ddl_transaction!
 
   # Dummy classes, to make migration possible across version changes
-  class Account < ApplicationRecord
-    has_one :user, inverse_of: :account
-    has_one :notification_policy, inverse_of: :account
-  end
-
   class User < ApplicationRecord
-    belongs_to :account
+    belongs_to :notification_policy, foreign_key: 'account_id', optional: true, inverse_of: false
```

Comment:
Does this also need a `primary_key` on it?

As written, I would think this is going to try to find `NotificationPolicy` records where the `users.account_id` value equals the `notification_policies.id` value ... maybe I'm missing something obvious here.

Reply:
Good catch! You're right, this needs `primary_key: 'account_id'`, fixing that.